src/main/scala/com/advancedtelematic/campaigner/actor/CampaignScheduler.scala

```diff
@@ -54,9 +57,10 @@ class CampaignScheduler(director: DirectorClient,
   def receive: Receive = {
     case NextBatch =>
       log.debug("Requesting next batch")
-      // TODO (OTA-2383) limit or stream device IDs from DB
-      campaigns.requestedDevices(campaign.id)
-        .map(deviceIds => BatchToSchedule(deviceIds.take(batchSize)))
+      campaigns.requestedDevicesStream(campaign.id)
```

Comment:
Is not some special statement/connection configuration required to configure a page size jdbc client should use? I can imagine default behaviour will be for driver fetch all the data and than Slick will turn it into a stream

Reply:
Hmm, though there is no mention of it in Akka Streams docs, it seems like you're right: https://mariadb.com/kb/en/library/about-mariadb-connector-j/#streaming-result-sets. Will fix it.